test/topology_experimental_raft/test_topology_upgrade.py

```diff
@@ -11,10 +11,86 @@
 
 from test.pylib.rest_client import HTTPError
 from test.pylib.manager_client import ManagerClient
-from test.pylib.util import wait_for_cql_and_get_hosts
+from test.pylib.util import read_barrier, wait_for_cql_and_get_hosts
 from test.topology.util import log_run_time, wait_until_topology_upgrade_finishes, \
         check_system_topology_and_cdc_generations_v3_consistency
+from cassandra.cluster import ConsistencyLevel
 
+def auth_data():
+    return [
+        {
+            "query": "INSERT INTO system_auth.roles (role, can_login, is_superuser, member_of, salted_hash) VALUES (?, ?, ?, ?, ?)",
```

Comment:
Can't we make the test more high level, i.e. no based on the representation of roles etc. in system tables, but based on the actual create role statements etc. -- as the user of the cluster would use them?

Perhaps inspecting the data directly in system tables is valuable too, but we should also have the test check that auth related queries (create role etc.) post-upgrade observe whatever state was introduced pre-upgrade